### ISSUE TITLE
Add Dart backend

### ISSUE THREAD

**Issue by Scorpiion** [blockforest/stagexlhand#9]
Hi, really cool repo!

I saw the "add zend backend" issue, and thought a Dart backend would be even better, maybe it's possible to do both.

**Comment by nilsdoehring** [comment]
Thanks!
I indeed plan to ultimately use a pure Dart backend. I'll just do the Zend backend first – because actually, it already exists thanks to the framework's ActionScript ancestor :-)

By the way, features are: persistence for all kinds of social functionality (users, friends, groups, user generated content including upload, invites, fangating, leaderboards, scores, levels, ...) and a user/content administration frontend (blacklisting, banning, etc). It was used in dozens of campaigns with millions of users and zero breaches.

**Comment by Scorpiion** [comment]
Okay, yeah that sounds like a good plan! 

Is this [rockdot](https://github.com/blockforest/rockdot) project the ancestor?

If you need any help with a Dart backend in the future, let me know! :)

**Comment by nilsdoehring** [comment]
Yes, that's the ancestor, not maintained anymore for obvious reasons. Same features, basically, plus compiling into AIR apps for iOS and Android. The Zend stuff is there, too. I developed it during my agency times, and it was used for clients such as Mercedes, Nike, Nikon, Bosch... in some way, they sponsored this :-)

After deciding for Dart/StageXL as successor platform, I was able to take over several hundred classes automatically by coming up with this: https://github.com/blockforest/stagexl-converter-pubglobal
(of course, after conversion, adaptions have to be made, but the syntax is correct enough to have your IDE guide you through this; worked way better than the AST-based conversion I tried first).

Thanks for offering assistance, I actually had a peek at your repo and noticed vane, which I will look into at least for inspiration – will get back to you :-) by the way, the url from the project description is leading nowhere.

**Comment by Scorpiion** [comment]
I see, cool clients indeed, "during my agency times", if you don't mind me asking, does that mean you switched workplace since then or more of a switch in what role at the same agency/company?

It's interesting that there are quite some ActionScript folks coming to Dart, I've been following Dart for a long time and the number of developers with an ActionScript background has surprised me to be honest, but when you look at ActionScript and that platform as whole it makes sense as a good successor platform as you said.

Yeah, sure, no problem, good that you mention it actually. I work fulltime coding in two languages, those are Dart and Go. Dart for REST and web parts and Go for more low/system level stuff. I've been a little bad at updating the Vane repo, it's been developed mostly internally the last couple of months, not huge changes but still. I should try to get it more up to date the coming months hopefully, some other stuff with higher priority atm. Don't hesitate to get in contact in the future, happy to help.

Ps. I fixed/removed the link now, thanks. Ds.
